subprojects/resources-s3/src/main/java/org/gradle/internal/resource/transport/aws/s3/S3Client.java

```diff
@@ -187,7 +187,7 @@ private String extractDirectoryName(String key) {
         Matcher matcher = DIRNAME_PATTERN.matcher(key);
         if (matcher.find()) {
             // Add a "/" to differentiate directories from files
-            return matcher.group(1) + "/";
+            return matcher.group(0);
```

Comment:
The comment does not make much sense now, does it? Please change it or remove it.